component/block_cache/block_cache.go

```diff
@@ -1119,12 +1136,22 @@ func (bc *BlockCache) upload(item *workItem) {
 	flock.Lock()
 	defer flock.Unlock()
 
+	endIndex := item.block.endIndex
+	// if a block has data less than block size and is not the last block,
+	// add null at the end and upload the full block
+	if item.block.endIndex-item.block.offset < bc.blockSize && item.block.endIndex < uint64(item.handle.Size) {
+		endIndex = item.block.offset + bc.blockSize
```

Comment:
We shall update the actual block.endindex as well as now the block is full for us.

Reply:
Done